src/dc_receive_imf.rs

```diff
@@ -642,7 +642,10 @@ unsafe fn add_parts(
                         }
                     }
                     if part.type_0 == Viewtype::Text {
-                        let msg_raw = CString::yolo(part.msg_raw.as_ref().unwrap().clone());
+                        let msg_raw = CString::yolo(match part.msg_raw.as_ref() {
+                            Some(x) => x.clone(),
+                            None => "".into(),
+                        });
```

Comment:
alternative could have been 

```rust
let msg_raw = CString::yolo(part.msg_raw.cloned().unwrap_or_default())
```

Reply:
so `""` is the default for a string? and `unwrap_or_default()` is the same as my initial `unwrap_or(&String::from("")).clone())`

Reply:
`"".to_string()` is the default for a string

Reply:
- `unwrap_or_default` is the same as `unwrap_or_else(|| Default::default())` 
- `.cloned()` is the same as `.map(|v| v.clone())`

Reply:
fine with @dignifiedquire's version as well.

Reply:
finally worked with `CString::yolo(part.msg_raw.as_ref().cloned().unwrap_or_default());` thanks all!